### DISCUSSION TITLE
Erro ao compilar para IOS

### DISCUSSION THREAD

**Discussion by saviobraz** [codemagic-ci-cd/codemagic-docs#2624] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Simplesmente estou seguindo todos os passos conforme a documentação para compilar para IOS, mas ocorre este erro 

### Expected results

== Building for iOS ==

> xcode-project use-profiles
Configure code signing settings
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.mobileprovision
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.provisionprofile
List available code signing certificates in keychain /Users/builder/Library/codemagic-cli-tools/keychains/04-07-24_8jc7atkn.keychain-db
Searching for files matching /Users/builder/clone/ios/**/*.xcodeproj
Failed to set code signing settings for ._Runner.xcodeproj

Build failed :|
Failed to set up code signing settings on Xcode projects

### Actual results

== Building for iOS ==

> xcode-project use-profiles
Configure code signing settings
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.mobileprovision
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.provisionprofile
List available code signing certificates in keychain /Users/builder/Library/codemagic-cli-tools/keychains/04-07-24_8jc7atkn.keychain-db
Searching for files matching /Users/builder/clone/ios/**/*.xcodeproj
Failed to set code signing settings for ._Runner.xcodeproj

Build failed :|
Failed to set up code signing settings on Xcode projects

### Build id (optional)

6686a71dfe3b1d5c22cb5dc8

**Comment by dtrdic** [comment]
hi @saviobraz i can see this error message inside logs 
`invalid argument: --xcode-project : '._Runner.xcodeproj' is not a valid Xcode project (OptionParser::InvalidArgument)`
something is messed up with your app. Try running `flutter create .` locally, let's see if regenerating the iOS project files can resolve this issue.
Try also by cleaning up your project with this commands;
```
cd ios
pod cache clean --all
rm -rf ~/Library/Caches/CocoaPods
rm -rf Pods
rm -rf ~/Library/Developer/Xcode/DerivedData/*
pod deintegrate
flutter clean
pod install
```

**Reply by saviobraz** [reply]
Ok amigo vou verificar aqui. Obrigado

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.

**Comment by hussainhabib2** [comment]
@dtrdic Can we please reopen this? I am facing exact same issue. I also tried flutter.create and the commands you mentioned. The app is being built perfectly fine on my local machine iOS simulator but giving this error on codemagic:


```

== Building for iOS ==

> xcode-project use-profiles
Configure code signing settings
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.mobileprovision
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.provisionprofile
List available code signing certificates in keychain /Users/builder/Library/codemagic-cli-tools/keychains/08-09-24_77mz2ol6.keychain-db
Searching for files matching /Users/builder/clone/ios/**/*.xcodeproj
Failed to set code signing settings for ._Runner.xcodeproj
Build failed :|
Failed to set up code signing settings on Xcode projects

```

This only happens when I enable iOS code signing in Distribution of Flutter workflow editor, otherwise codemagic builds ios just fine.
